### PR TITLE
critcl: init at 3.2, use in tcllib

### PR DESCRIPTION
**Pull request by fgaz**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
looks good. some nits:
pkg/by-name ?

File: pkgs/development/compilers/critcl/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, tcl
+, tcllib
+}:
+
+tcl.mkTclDerivation rec {
+  pname = "critcl";
+  version = "3.2";
```
[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/development/compilers/critcl/default.nix:10]: the `rec` version makes it easier to convert this to `finalAttrs:` when I get around to add the feature to `mkTclDerivation`

File: pkgs/development/compilers/critcl/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = fetchFromGitHub {
+    owner = "andreas-kupries";
+    repo = "critcl";
+    rev = version;
+    hash = "sha256-IxScn9ZTlqD9mG9VJLG+TtplLFhhahOiFhQCjxp22Uk=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
```
[the remark this answers is not in the document]
Comment (fgaz) [reply pkgs/development/compilers/critcl/default.nix]: I just forgot to remove it. Done.

**Comment by fgaz** [comment]
> pkg/by-name ?

not possible because of the `callPackage` arguments

**Pull request merged**